pkg/utils/controller/handlers.go

```diff
@@ -21,32 +24,53 @@ func AddEventHandlers(informer cache.SharedInformer, a addFunc, u updateFunc, d
 	})
 }
 
+func AddKeyedEventHandlers(logger logr.Logger, informer cache.SharedInformer, queue workqueue.RateLimitingInterface, keyFunc keyFunc) {
+	AddEventHandlers(informer, AddFunc(logger, queue, keyFunc), UpdateFunc(logger, queue, keyFunc), DeleteFunc(logger, queue, keyFunc))
+}
+
 func AddDefaultEventHandlers(logger logr.Logger, informer cache.SharedInformer, queue workqueue.RateLimitingInterface) {
-	AddEventHandlers(informer, Add(logger, queue), Update(logger, queue), Delete(logger, queue))
+	AddKeyedEventHandlers(logger, informer, queue, MetaNamespaceKey)
 }
 
-func Enqueue(logger logr.Logger, queue workqueue.RateLimitingInterface, obj interface{}) {
-	if key, err := cache.MetaNamespaceKeyFunc(obj); err != nil {
-		logger.Error(err, "failed to compute key name")
+func Enqueue(logger logr.Logger, queue workqueue.RateLimitingInterface, obj interface{}, keyFunc keyFunc) {
+	if key, err := keyFunc(obj); err != nil {
+		logger.Error(err, "failed to compute key name", "obj", obj)
 	} else {
 		queue.Add(key)
 	}
 }
 
-func Add(logger logr.Logger, queue workqueue.RateLimitingInterface) addFunc {
+func MetaNamespaceKey(obj interface{}) (interface{}, error) {
+	return cache.MetaNamespaceKeyFunc(obj)
+}
+
+func Explicit[K any](keyFunc func(K) cache.ExplicitKey) keyFunc {
```

Comment:
Can we rename the function's arg `keyFunc`? It's quite confusing with the type `keyFunc`. 

Something like this:

```
func Explicit[K any](parseKey func(K) cache.ExplicitKey) keyFunc {}
```

Reply:
Changed.